File: src/decoders/payment-method.ts
import { PaymentMethod, UnknownEnumValue } from "~/models";

export const decodePaymentMethod = (value: any): PaymentMethod => {
  const unwrapped = value.name;

  switch (unwrapped) {
    case "Carte bancaire":
      return PaymentMethod.CARD;
    case "Espèce":
      return PaymentMethod.CASH;
    case "IZLY":
      return PaymentMethod.IZLY;
    case "Monéo":
      return PaymentMethod.MONEO;
    default:
      throw new UnknownEnumValue("PaymentMethod", unwrapped);
  }
};

File: library/src/commonMain/kotlin/ink/literate/crowous/decoders/payment-method.kt
package ink.literate.crowous.decoders

import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonPrimitive
import ink.literate.crowous.models.PaymentMethod

fun decodePaymentMethod (method: JsonObject): PaymentMethod {
  return when (val value = method["name"]!!.jsonPrimitive.content) {
    "Carte bancaire" -> PaymentMethod.CARD
    "Espèce" -> PaymentMethod.CASH
    "IZLY" -> PaymentMethod.IZLY
    "Monéo" -> PaymentMethod.MONEO
    else -> throw Exception("Unknown payment method: $value")
  }
}
